### ISSUE TITLE
sum all logits in a batch

### ISSUE THREAD

**Issue by jbjeong91** [TaoYang225/AD-DROP#2]
Why did you sum all logits in a batch to get gradient?

line number 47 in trainer.py ;
outputs_true = outputs_ori.gather(1, p_label.view(-1, 1)).sum() _-> sum all logits in a batch_

you use 'outputs_true' to get gradient like this ;
(grad,) = torch.autograd.grad(outputs_true.view(-1), attention_ori[layer], retain_graph=True)

**Comment by TaoYang225** [comment]
Hi, this is similar to computing the gradient of the batch loss. For each sample, summing the logits and then computing the gradient is equivalent to computing the gradient for each individual logit separately.
